packages/plugins/typescript/react-apollo/src/visitor.ts

```diff
@@ -583,4 +589,58 @@ export class ReactApolloVisitor extends ClientSideBaseVisitor<
       .filter(a => a)
       .join('\n');
   }
+
+  public get fragments(): string {
+    const fragments = super.fragments;
+
+    if (this._fragments.length === 0 || !this.config.withFragmentHooks) {
+      return fragments;
+    }
+
+    const operationType = 'Fragment';
+
+    const hookFns: string[] = [fragments];
+
+    for (const fragment of this._fragments.values()) {
+      if (fragment.isExternal) {
+        continue;
+      }
+
+      const nodeName = fragment.name ?? '';
+      const suffix = this._getHookSuffix(nodeName, operationType);
+      const fragmentName: string =
+        this.convertName(nodeName, {
+          suffix,
+          useTypesPrefix: false,
+          useTypesSuffix: false,
+        }) + this.config.hooksSuffix;
+
+      const operationTypeSuffix: string = this.getOperationSuffix(fragmentName, operationType);
+
+      const operationResultType: string = this.convertName(nodeName, {
+        suffix: operationTypeSuffix + this._parsedConfig.operationResultSuffix,
+      });
+
+      const IDType = this.scalars.ID ?? 'string';
+
+      const hook = `export function use${fragmentName}(id: ${IDType}) {
```

Comment:
I'm also interested in this but my usecase involves custom keyFields, would be great if the input type can be inferred from imported type policies or be a generic object instead like:

```suggestion
      const hook = `export function use${fragmentName}<F = { id: ${IDType} }>(from: F) {
```

inferring from type policy can also throw type errors on invalid use of `useFragment`

Reply:
Added support for `keyFields`, thanks for the suggestion!